core/actor/src/test/scala/net/liftweb/actor/LAFutureSpec.scala

```diff
@@ -61,30 +61,81 @@ class LAFutureSpec extends Specification {
       result shouldEqual givenFailure
     }
 
-    "collect one future result" in {
-      val givenOneResult = 123
-      val one = LAFuture(() => givenOneResult)
-      LAFuture.collect(one).get(timeout) shouldEqual List(givenOneResult)
-    }
+    "when collecting results with LAFuture.collect" in {
+      "collect one future result" in {
+        val givenOneResult = 123
+        val one = LAFuture(() => givenOneResult)
+        LAFuture.collect(one).get(timeout) shouldEqual List(givenOneResult)
+      }
 
-    "collect more future results in correct order" in {
-      val givenOneResult = 123
-      val givenTwoResult = 234
-      val one = LAFuture(() => givenOneResult)
-      val two = LAFuture(() => givenTwoResult)
-      LAFuture.collect(one, two).get(timeout) shouldEqual List(givenOneResult, givenTwoResult)
-    }
+      "collect more future results in correct order" in {
+        val givenOneResult = 123
+        val givenTwoResult = 234
+        val one = LAFuture(() => givenOneResult)
+        val two = LAFuture(() => givenTwoResult)
+        LAFuture.collect(one, two).get(timeout) shouldEqual List(givenOneResult, givenTwoResult)
+      }
+
+      "collect empty list immediately" in {
+        val collectResult = LAFuture.collect(Nil: _*)
+        collectResult.isSatisfied shouldEqual true
+        collectResult.get(timeout) shouldEqual Nil
+      }
 
-    "collect empty list immediately" in {
-      val collectResult = LAFuture.collect(Nil: _*)
-      collectResult.isSatisfied shouldEqual true
-      collectResult.get(timeout) shouldEqual Nil
+      "report a failed LAFuture as a failure for the overall future" in {
+        val one: LAFuture[Int] = new LAFuture
+        val two: LAFuture[Int] = LAFuture(() => 5)
+
+        one.fail(Failure("boom boom boom!"))
+
+        val collectResult = LAFuture.collect(one, two)
+        collectResult.get(timeout) shouldEqual Failure("boom boom boom!")
+      }
     }
 
-    "collectAll empty list immediately" in {
-      val collectResult = LAFuture.collectAll(Nil : _*)
-      collectResult.isSatisfied shouldEqual true
-      collectResult.get(timeout) shouldEqual Nil
+    "when collecting Boxed results with collectAll" in {
+      "collectAll collects an EmptyBox immediately" in {
+        val one: LAFuture[Box[Int]] = LAFuture(() => { Failure("whoops"): Box[Int] })
+        val two: LAFuture[Box[Int]] = LAFuture(() => { Thread.sleep(10000); Full(1) })
+
+        val collectResult = LAFuture.collectAll(one, two)
+        collectResult.get(5000) shouldEqual Failure("whoops")
```

Comment:
Per my response above, we could but we shouldn't. `collectAll` doesn't guarantee the *execution* of any of the futures---in fact, it doesn't affect their execution at all. All it does is gather their results. So if there were a way to furnish a future that would never get scheduled, `collectAll` would simply never complete.

Of course for Lift futures the act of creating the future schedules it, but basically that's orthogonal to this code. So I don't think we should test it here.